nlp/src/main/java/org/mifos/chatbot/nlp/RasaNLUService.java

```diff
@@ -0,0 +1,113 @@
+@Slf4j
+public class RasaNLUService implements NLPService {
+
+    @Value("${rasa.nlu.service.url}")
+    String url;
+
+    @Override
+    public Intent recognize(String input) throws IOException {
+        JSONObject payload = createPayload(input);
+        HttpResponse rasaNLUResponse = getResponseFromRasaNLU(payload);
+        String result = getResult(rasaNLUResponse);
+        JSONObject resultJSON;
+        try {
+            JSONParser jsonParser = new JSONParser();
+            resultJSON = (JSONObject) jsonParser.parse(result);
+        } catch (ParseException e) {
+            log.error(e.toString());
+            resultJSON = null;
+        }
+            return getIntent(resultJSON, input);
+    }
+
+    private Intent getIntent(JSONObject resultJSON, String input) {
+        if (resultJSON == null) {
+            return null;
+        }
+        List<Intent> resultIntents = new ArrayList<>();
+        String subIntent = resultJSON.get("intent").toString().split("\"")[5];
+        resultIntents.add(new Intent(subIntent));
+        Intent intent = null;
+        for (Intent resultIntent : resultIntents) {
+            intent = resultIntent;
+            intent.addParameter("ID", findId(input));
+            intent.addParameter("Date", findDate(input));
+        }
+        log.info("Found {} intents", intent.getKeyword());
+        return intent;
+    }
+
+    private JSONObject createPayload(String input) {
+        JSONObject payload = new JSONObject();
+        payload.put("text", input);
+        return payload;
+    }
+
+    private HttpResponse getResponseFromRasaNLU(JSONObject payload) throws IOException {
+        HttpClient client = new DefaultHttpClient();
+        HttpPost post = new HttpPost(url);
+        post.setEntity(new StringEntity(payload.toString(), ContentType.APPLICATION_JSON));
+        return client.execute(post);
+    }
+
+    private String getResult(HttpResponse RasaNLUResponse) throws IOException {
+        BufferedReader rd = new BufferedReader(
+                new InputStreamReader(RasaNLUResponse.getEntity().getContent()));
+
+        StringBuilder result = new StringBuilder();
+        String line = "";
+        while ((line = rd.readLine()) != null) {
+            result.append(line);
+        }
+        log.info("Result from RasaNLU {}", result.toString());
+        return result.toString();
+    }
+
+    private String findDate(String input) throws IndexOutOfBoundsException {
+        try {
+            List<Date> dates = new Parser().parse(input).get(0).getDates();
+            Date date = dates.get(0);
+            return (date.getMonth() + 1) + "-" + date.getDate() + "-" + (date.getYear() + 1900);
+        } catch (Exception e) {
+            return "";
+        }
+    }
+
+    private long findId(String input) {
+        long id = 1L;
+        for (String token : input.split(" ")) {
+            try {
+                id = Long.parseLong(token);
+            } catch (NumberFormatException e) {
```

Comment:
Should be handled, ideally. If this isn't expected, you should trigger some sort of alerting.